spec/03-plugins/18-request-termination/02-access_spec.lua

```diff
@@ -0,0 +1,177 @@
+local helpers = require "spec.helpers"
+local cache = require "kong.tools.database_cache"
+local cjson = require "cjson"
+
+describe("Plugin: request-termination (access)", function()
+  local plugin_config
+  local client, admin_client
+
+  setup(function()
+    local api1 = assert(helpers.dao.apis:insert {
+      name = "api-1",
+      hosts = { "api1.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api1.id,
+      config = {
+      }
+    })
+    local api2 = assert(helpers.dao.apis:insert {
+      name = "api-2",
+      hosts = { "api2.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api2.id,
+      config = {
+        status_code=404
+      }
+    })
+    local api3 = assert(helpers.dao.apis:insert {
+      name = "api-3",
+      hosts = { "api3.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api3.id,
+      config = {
+        status_code=406,
+        message="Invalid"
+      }
+    })
+    local api4 = assert(helpers.dao.apis:insert {
+      name = "api-4",
+      hosts = { "api4.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api4.id,
+      config = {
+        body="<html><body><h1>Service is down for maintenance</h1></body></html>"
+      }
+    })
+    local api5 = assert(helpers.dao.apis:insert {
+      name = "api-5",
+      hosts = { "api5.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api5.id,
+      config = {
+        status_code=451,
+        content_type="text/html",
+        body="<html><body><h1>Service is down due to content infringement</h1></body></html>"
+      }
+    })
+    local api6 = assert(helpers.dao.apis:insert {
+      name = "api-6",
+      hosts = { "api6.request-termination.com" },
+      upstream_url = "http://mockbin.com"
+    })
+    assert(helpers.dao.plugins:insert {
+      name = "request-termination",
+      api_id = api6.id,
+      config = {
+        status_code=503,
+        body='{"code": 1, "message": "Service unavailable}'
```

Comment:
there is a missing quote at the end. Test passes, but is this intentional? if so a small comment as to why would make sense.